builder.go

```diff
@@ -12,21 +12,20 @@ type ErrorBuilder struct {
 	errorType     *Type
 	message       string
 	cause         error
-	underlying    []error
 	mode          callStackBuildMode
 	isTransparent bool
 }
 
 // NewErrorBuilder creates error builder from an existing error type.
-func NewErrorBuilder(t *Type) *ErrorBuilder {
+func NewErrorBuilder(t *Type) ErrorBuilder {
```

Comment:
ErrorBuilder is routinely copied several times, are you sure that value receiver is an improvement here?

Reply:
Oh, I missed a benchmark in commit message. OK. 
Well, this is sad, actually. Go compiler fails to perform an escape analysis, so it is cheaper to risk extra copying.